### PR TITLE
Add commands for testing apis

### PR DESCRIPTION
**Pull request by RobertLucian**
Add test commands for APIs.

---

checklist:

- [x] run `make test` and `make lint`
- [x] test manually the commands

### CODE REVIEW

File: build/test.sh [hunk 1]
```diff
@@ -28,12 +72,20 @@ function run_python_tests() {
...
 }
 
-cmd=${1:-""}
+function run_api_tests() {
+  if [ "$provider" = "aws" ]; then
+    pytest $ROOT/test/e2e/tests -k aws --aws-env "$cluster_env"
+  elif [ "$provider" = "gcp" ]; then
+    pytest $ROOT/test/e2e/tests -k gcp --gcp-env "$cluster_env"
+  fi
+}
```
Comment (miguelvr) [inline build/test.sh:81]: can we call this `run_e2e_tests` instead? for consistency

File: build/test.sh [hunk 2]
```diff
@@ -28,12 +72,20 @@ function run_python_tests() {
...
 
 if [ "$cmd" = "go" ]; then
   run_go_tests
 elif [ "$cmd" = "python" ]; then
   run_python_tests
+elif [ "$cmd" = "apis" ]; then
+  run_api_tests
```
Comment (miguelvr) [inline build/test.sh:88]: same here: `apis` -> `e2e`

File: Makefile [hunk 3]
```diff
@@ -258,6 +259,14 @@ test-go:
 test-python:
 	@./build/test.sh python
 
+test-e2e-existing-clusters:
+	@eval $$(python3 ./manager/cluster_config_env.py ./dev/config/cluster-aws.yaml) && CORTEX_CLI_PATH=./bin/cortex ./build/test.sh apis -p aws -e "$$CORTEX_CLUSTER_NAME-aws"
+	@eval $$(python3 ./manager/cluster_config_env.py ./dev/config/cluster-gcp.yaml) && CORTEX_CLI_PATH=./bin/cortex ./build/test.sh apis -p gcp -e "$$CORTEX_CLUSTER_NAME-gcp"
```
Comment (miguelvr) [inline Makefile:264]: would be useful to run these in parallel. Otherwise, you can just run a single command

File: Makefile [hunk 4]
```diff
@@ -258,6 +259,14 @@ test-go:
...
+test-e2e-existing-cluster-aws:
+	@eval $$(python3 ./manager/cluster_config_env.py ./dev/config/cluster-aws.yaml) && CORTEX_CLI_PATH=./bin/cortex ./build/test.sh apis -p aws -e "$$CORTEX_CLUSTER_NAME-aws"
+test-e2e-existing-cluster-gcp:
+	@eval $$(python3 ./manager/cluster_config_env.py ./dev/config/cluster-gcp.yaml) && CORTEX_CLI_PATH=./bin/cortex ./build/test.sh apis -p gcp -e "$$CORTEX_CLUSTER_NAME-gcp"
```
Comment (miguelvr) [inline Makefile:268]: nit: command naming is a bit too long

**Review by miguelvr — CHANGES_REQUESTED** [review]

File: build/test.sh (hunk 1 above)
Comment (RobertLucian) [reply build/test.sh:81]: Yep, I agree.

File: Makefile (hunk 4 above)
Comment (RobertLucian) [reply Makefile:268]: Not too concerned about this - there are commands longer than this in Makefile.

Comment (RobertLucian) [reply Makefile]: We want to go with `test-e2e-aws/gcp` for existing clusters and `test-e2e-new-aws/gcp` for new clusters to be spun up.

**Review by deliahu — APPROVED** [review]

**Pull request merged**